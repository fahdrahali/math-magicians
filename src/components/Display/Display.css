.input {
    grid-column: 1 / 5;
    outline: none;
    background-color: #b4b1b1;
    border: none;
    height: 60px;
    font-size: 30px;
    color: #ffff;
    padding: 10px;
    text-align: right;
  }