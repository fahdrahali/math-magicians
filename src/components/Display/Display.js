import { checkPropTypes } from 'prop-types';
import React from 'react';
import './Display.css';

const Display = (props) => {
  const { input } = props;
  return (
    <input
      className="input"
      type="text"
      value={`${input.total === null ? '' : input.total}${
        input.operation === null || input.operation === undefined ? '' : input.operation
      }${input.next === null ? '' : input.next}`}
      disabled
    />
  );
};

Display.propTypes = {
  input: checkPropTypes.isRequired,
};
export default Display;
