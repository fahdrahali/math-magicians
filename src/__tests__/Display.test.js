import React from 'react';
import TestRenderer from 'react-test-renderer';
import Display from '../components/Display';

it('renders correctly', () => {
  const renderer = TestRenderer.create(
    <Display input={{ total: 0, next: '', operation: '' }} />,
  ).toJSON();
  expect(renderer).toMatchSnapshot(
    `
      <input
       className="input"
       disabled={true}
       type="text"
       value="0"
     />
      `,
  );
  // console.log(renderer);
});
