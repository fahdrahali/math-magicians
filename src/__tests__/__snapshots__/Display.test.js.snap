// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly: 
      <input
       className="input"
       disabled={true}
       type="text"
       value="0"
     />
       1`] = `
<input
  className="input"
  disabled={true}
  type="text"
  value="0"
/>
`;
